src/error.js

```diff
@@ -195,13 +207,17 @@ function reportErrorToServer(message, filename, line, col, error) {
  */
 export function getErrorReportUrl(message, filename, line, col, error,
     hasNonAmpJs) {
+  let expected = false;
   if (error) {
     if (error.message) {
       message = error.message;
     } else {
       // This should never be a string, but sometimes it is.
       message = String(error);
     }
+    if (error.expected) {
```

Comment:
Added docs here and elsewhere inline. But do we have any overall documentation for errors in our repo?